R/aggregate_kicking_stats.R

```diff
@@ -0,0 +1,161 @@
+#' Summarize Kicking Stats
+#'
+#' @description Build columns that aggregate kicking stats at the game level.
+#'
+#' @param pbp A Data frame of NFL play-by-play data typically loaded with
+#' [load_pbp()] or [build_nflfastR_pbp()].
+#' @param weekly If `TRUE`, returns week-by-week stats, otherwise, stats for
+#' the entire Data frame.
+#'
+#' @examples
+#'
+#' \donttest{
+#' pbp <- nflreadr::load_pbp(2021)
+#' calculate_kicking_stats(pbp, weekly = TRUE)
+#' }
+#'
+#' @return a dataframe of kicking stats
+#' @seealso <https://nflreadr.nflverse.com/reference/load_player_stats.html> for the nflreadr function to download this from repo (`stat_type = "kicking"`)
+#' @export
+calculate_kicking_stats <- function(pbp, weekly = FALSE){
+
+  # First, creating a grouping variable object to toggle the weekly argument w/
+  grp_vars <- list("season", "season_type", "week", "team", "player_id")
```

Comment:
Hmm. This is a bit inconsistent with how the other two aggregate functions work - basically, if weekly = FALSE, it would group by team + player_id (and give a summary over the entire pbp that is passed into the function)

Reply:
In clearer terms, you could probably go with `grp_vars <- c("player_id", "team")` for `!weekly`.

Reply:
So, initially I changed it and merged on season (for weekly = `TRUE` or `FALSE`) and season_type (for weekly = `TRUE`) later. But, I changed it back because I thought it was easier to just include them in the group_by statement. Happy to discuss / explain the rationale a bit more.